src/foam/nanos/controller/ApplicationController.js

```diff
@@ -211,7 +211,7 @@ foam.CLASS({
       class: 'foam.u2.ViewSpec',
       name: 'loginView',
       expression: function(theme$loginView) {
-        return theme$loginView
+        return JSON.parse(theme$loginView);
```

Comment:
ViewSpec has an adapt that already does this, not sure why it doesnt work here

Reply:
it might work and the default value for themeloginview was wrong since it was just a string of the classid instead of string of the viewspec? will try in a sec

Reply:
Dont think we would, we do the same thing for top and side nav too, we keep ViewSpecs as Strings in theme cause jrls dont support ViewSpecs

Reply:
ok will check those then, thanks

Reply:
works now after setting from useCustomElements, the adapt wasn't called when setting from the expression